src/Routers/AppRouter.tsx

```diff
@@ -24,11 +24,26 @@ import HCXRoutes from "./routes/HCXRoutes";
 import ShiftingRoutes from "./routes/ShiftingRoutes";
 import AssetRoutes from "./routes/AssetRoutes";
 import ResourceRoutes from "./routes/ResourceRoutes";
-import { DetailRoute } from "./types";
+import { usePluginRoutes } from "@/Common/hooks/useCareApps";
 import careConfig from "@careConfig";
 import IconIndex from "../CAREUI/icons/Index";
 
-const Routes = {
+export type RouteParams<T extends string> =
+  T extends `${string}:${infer Param}/${infer Rest}`
+    ? { [K in Param | keyof RouteParams<Rest>]: string }
+    : T extends `${string}:${infer Param}`
+      ? { [K in Param]: string }
+      : Record<string, never>;
+
+export type RouteFunction<T extends string> = (
+  params: RouteParams<T>,
+) => JSX.Element;
+
+export type AppRoutes = {
+  [K in string]: RouteFunction<K>;
+};
```

Comment:
Does Raviger have route types that we can borrow instead of redefining types?

Reply:
The existing type usage was ugly. Unlikely that raviger has something that enforces types that enforces defining all path params. 

@shivankacker Do find out and update here. If it's not there, consider making a PR to raviger. If so, this could be a good addition to raviger.

Reply:
I checked and I could not find something strongly typed (Best I could find was the `Route` type but you will have to manually give the types for params in that)
I can make a PR to raviger, just skimmed through its codebase yesterday. For now our present types are great